### PR TITLE
ILMS-288 - fix conditions - don't sent device if resource is off

### PR DESCRIPTION
**Pull request by Channa53**
### PR Info
- [x] Please functional test this PR
- [ ] Already tested in KMS and stand alone

### CODE REVIEW

File: src/components/Settings/Settings.tsx [hunk 1]
```diff
@@ -157,16 +157,20 @@ export class Settings extends Component<Props, State> {
...
                 ? device
                 : this.getCurrentDevice(ResourceTypes.AUDIO);
 
-        onSettingsChanged(true, false, screenShareOn, camera, audio);
+        onSettingsChanged(
+            true,
+            false,
+            screenShareOn,
+            cameraOn ? cameraDevice : undefined,
+            audioOn ? audioDevice : undefined
```
Comment (tomm1990) [inline src/components/Settings/Settings.tsx:172]: its possible to search for a `cameraDevice` and send `undefined` to **onSettingsChanged** function if `cameraOn = false`.
you should change that logic.

**Review by tomm1990 — CHANGES_REQUESTED** [review]

File: src/components/Settings/Settings.tsx (hunk 1 above)
Comment (Channa53) [reply src/components/Settings/Settings.tsx:170]: you are not able to choose device if camera is off. thats why I check device.kind

File: src/components/Settings/Settings.tsx [hunk 2]
```diff
@@ -157,16 +157,20 @@ export class Settings extends Component<Props, State> {
     handleChooseDevice = (device: MediaDeviceInfo) => {
```
[the remark this answers is not in the document]
Comment (Channa53) [reply src/components/Settings/Settings.tsx:170]: done

File: src/components/Settings/Settings.tsx (hunk 1 above)
Comment (Channa53) [reply src/components/Settings/Settings.tsx]: but it changed by the recent commit

**Review by tomm1990 — APPROVED** [review]

**Pull request merged**